packages/aws-cdk-lib/aws-ec2/README.md

```diff
@@ -1547,6 +1547,36 @@ const aspect = new ec2.InstanceRequireImdsv2Aspect();
 Aspects.of(this).add(aspect);
 ```
 
+### Associating a Public IP Address with an Instance
+
+All subnets have an attribute that determines whether instances launched into that subnet are assigned a public IPv4 address. This attribute is set to true by default for default public subnets. Thus, an EC2 instance launched into a default public subnet will be assigned a public IPv4 address. Nondefault public subnets have this attribute set to false by default and any EC2 instance launched into a nondefault public subnet will not be assigned a public IPv4 address automatically. To automatically assign a public IPv4 address to an instance launched into a nondefault public subnet, you can set the `associatePublicIpAddress` property on the `Instance` construct to true. Alternatively, to not automatically assign a public IPv4 address to an instance launched into a default public subnet, you can set `associatePublicIpAddress` to false. Importantly, if you set `associatePublicIpAddress` and there is a VPC defined in the same stack, you must make sure the `Instance` has a dependency on the VPC to declare a dependency on the VPC-gateway attachment.
+
+```ts
+const vpc = new ec2.Vpc(this, 'VPC', {
+  cidr: '10.0.0.0/16',
+  natGateways: 0,
+  maxAzs: 3,
+  subnetConfiguration: [
+    {
+      name: 'public-subnet-1',
+      subnetType: ec2.SubnetType.PUBLIC,
+      cidrMask: 24,
+    },
+  ],
+});
+
+const instance = new ec2.Instance(this, 'Instance', {
+  vpc,
+  vpcSubnets: { subnetGroupName: 'public-subnet-1' },
+  instanceType: ec2.InstanceType.of(ec2.InstanceClass.T3, ec2.InstanceSize.NANO),
+  machineImage: new ec2.AmazonLinuxImage({ generation: ec2.AmazonLinuxGeneration.AMAZON_LINUX_2 }),
+  detailedMonitoring: true,
+  associatePublicIpAddress: true,
+});
+
+instance.node.addDependency(vpc);
```

Comment:
I think you may be able to accomplish in this PR this by using the `internetConnectivityEstablished` dependable.

example
https://github.com/aws/aws-cdk/blob/53d24f1a1357410746052bbf25b5b89b985f7b80/packages/aws-cdk-lib/aws-rds/lib/cluster.ts#L878-L878

```ts
this.instance.node.addDependency(internetConnected);
```

Reply:
Good suggestion. I've added this into the instance constructor and updated the README to remove instance.node.addDependency(vpc).